daemon/algod/api/server/v2/handlers.go

```diff
@@ -984,6 +984,7 @@ type PreEncodedSimulateTxnResult struct {
 	LogicSigBudgetConsumed   *uint64                                 `codec:"logic-sig-budget-consumed,omitempty"`
 	TransactionTrace         *model.SimulationTransactionExecTrace   `codec:"exec-trace,omitempty"`
 	UnnamedResourcesAccessed *model.SimulateUnnamedResourcesAccessed `codec:"unnamed-resources-accessed,omitempty"`
+	TxnInitialState          *model.SimulateTxnInitialState          `codec:"txn-initial-state,omitempty"`
```

Comment:
I can see the temptation to attach an object to each Txn, but I think everything will get simpler if there's just a single object, up at the group level, called something like `InitialResourceState` and it's a complete catalog of the initial states of all resources that were touched during execution. The "initial" state should be the state before any transaction at all was executed.

By placing these objects down here, on transactions, I think you get two problems.
1. You probably end up with a lot of repetition
2. It's not obvious what "initial" means.  The state before the group, or the before the individual transaction.

Since we are sending every change along the way, I feel like we only need the true initial states, from before any txn is evaluated. (But I think the way we decide what to include is by evaluating the group and see what is touched.)